pkg/api/store/workload/workload_store.go

```diff
@@ -78,6 +78,11 @@ func (s *CustomizeStore) Update(apiContext *types.APIContext, schema *types.Sche
 	}
 	setScheduling(apiContext, data)
 	setStrategy(data)
+	if updated, err := imageUpdated(apiContext, data); err != nil {
```

Comment:
Is the main point of the fix is to trigger secretsUpdate on workload update? And by adding an extra check - imageUpdated - is just to guard against the condition when image doesn't need to be updated?

Pods get recreated when pod spec changes. The way k8s resource upgrade works in rancher - we do merged based approach, where fields from the update get merged with the existing fields. If we are passing exactly the same list of image pull secrets, no update should happen. Can you see if we really need an imageUpdated check because it does an extra call to the API - something we should avoid. First step to do after disabling the check would be - compare the specs (view yml for pre and post upgraded workload, and see what got changed in the pod spec. May be we order in image pull secret array causes the change?

Reply:
My concern was mostly because `imagePullSecrets` is a part of pod spec. 
If any extra secrets are added for the same url, then `imagePullSecrets` will get updated on workload update even if that secret isn't required (since we attach all of matching url secrets) 

I didn't know I could access the pre and post upgraded yaml, I can use that to decide if the update is necessary.. then api call won't be required, I'll update code to use this logic